feat(certificate-auth): Upgraded plugin to use golang v1.21

rvelaVenafi opened the pull request:
* Upgraded golang to v1.21

* Upgraded plugin dependencies to latest versions

* Fixed 224 linter issues

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Upgraded plugin to use golang v1.21

* Upgraded golang to v1.21

* Upgraded plugin dependencies to latest versions

* Fixed 224 linter issues

Closes VC-27221

luispresuelVenafi commented on go.mod:
```diff
@@ -1,18 +1,227 @@
 module github.com/terraform-providers/terraform-provider-venafi
 
-go 1.12
+go 1.21
```
This will fail as current pipeline is using default 1.17, make sure to update the pipeline script as well

rvelaVenafi replied:
got it

rvelaVenafi committed:
feat(certificate-auth): Upgraded plugin to use golang v1.21

* Upgraded golang to v1.21

* Upgraded plugin dependencies to latest versions

* Fixed 224 linter issues

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Upgraded plugin to use golang v1.21

* Upgraded golang to v1.21

* Upgraded plugin dependencies to latest versions

* Fixed 224 linter issues

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Upgraded plugin to use golang v1.21

* Upgraded golang to v1.21

* Upgraded plugin dependencies to latest versions

* Fixed 224 linter issues

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Downgraded to go v1.20

Looks like we have issues with golang v1.21; dongrading to v1.20

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Downgraded to go v1.20

Updated go.sum

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Back to go v1.21

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Updated make linter

Updated make linter target to use go install instead of golangci-lint install script

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Updated make linter

Used new script for golangci-lint

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Updated module name to match repository

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Updated imports after module renaming

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Fixed linter errors

Closes VC-27221

rvelaVenafi committed:
feat(certificate-auth): Updated test target

Updated test target to follow a more golang approach

Closes VC-27221
